conans/client/conan_api.py

```diff
@@ -275,7 +281,10 @@ def get_test_conanfile_path(tf):
                     raise ConanException("test folder '%s' not available, "
                                          "or it doesn't have a conanfile.py" % tf)
 
-        test_conanfile_path = get_test_conanfile_path(test_folder)
+        test_conanfile_path = None
+        if test_folder is not False:
```

Comment:
Why not put this inside the ``get_test_conanfile_path()`` function, that is where the logic for this arg is managed?

Reply:
I thought that it was clear to manage the added exception ``(test_folder == False)`` of disabling test outside of the nested function because it's basically returning the test path and behaving like before.

But I don't mind if you feel that is better to manage it inside of the nested function.

Reply:
The function also returns ``None`` if no test_folder is found, which is possible when the argument is ``None``, so the semantics of the function is not "always return a valid path". If you put it into the function, you get the management of ``test-folder`` tri-state argument in consecutive lines, which is nice.

Reply:
Fixing it!!!